CKAN command line 'compare' option unhandled exception

The CKAN command line 'compare' option drops an unhandled exception if called without arguments,

    dastardly{aa}: mono ckan.exe compare
    
    Unhandled Exception:
    System.ArgumentNullException: Argument cannot be null.
    Parameter name: input
      at System.Text.RegularExpressions.Regex.Match (System.String input) [0x00000] in <filename unknown>:0 
      at System.Text.RegularExpressions.Regex.Match (System.String input, System.String pattern, RegexOptions options, TimeSpan matchTimeout) [0x00000] in <filename unknown>:0 
      at System.Text.RegularExpressions.Regex.Match (System.String input, System.String pattern) [0x00000] in <filename unknown>:0 
      at CKAN.Version..ctor (System.String version) [0x00000] in <filename unknown>:0 
      at CKAN.CmdLine.Compare.RunCommand (CKAN.KSP ksp, System.Object rawOptions) [0x00000] in <filename unknown>:0 
      at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00000] in <filename unknown>:0 
    [ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentNullException: Argument cannot be null.
    Parameter name: input
      at System.Text.RegularExpressions.Regex.Match (System.String input) [0x00000] in <filename unknown>:0 
      at System.Text.RegularExpressions.Regex.Match (System.String input, System.String pattern, RegexOptions options, TimeSpan matchTimeout) [0x00000] in <filename unknown>:0 
      at System.Text.RegularExpressions.Regex.Match (System.String input, System.String pattern) [0x00000] in <filename unknown>:0 
      at CKAN.Version..ctor (System.String version) [0x00000] in <filename unknown>:0 
      at CKAN.CmdLine.Compare.RunCommand (CKAN.KSP ksp, System.Object rawOptions) [0x00000] in <filename unknown>:0 
      at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00000] in <filename unknown>:0 
    dastardly{aa}:

--- a/Cmdline/Action/Compare.cs
+++ b/Cmdline/Action/Compare.cs
@@ -2,16 +2,52 @@
 {
     public class Compare : ICommand
     {
+        private IUser user;
+
+        public Compare(IUser user)
+        {
+            this.user = user;
+        }
+
         public int RunCommand(CKAN.KSP ksp, object rawOptions)
         {
             var options = (CompareOptions)rawOptions;
 
-            var leftVersion = new Version(options.Left);
-            var rightVersion = new Version(options.Right);
+            if (options.Left != null && options.Right != null)
+            {
+                var leftVersion = new Version(options.Left);
+                var rightVersion = new Version(options.Right);
 
-            ksp.User.RaiseMessage(leftVersion.CompareTo(rightVersion).ToString());
+                int compareResult = leftVersion.CompareTo(rightVersion);
+                if (compareResult == 0)
+                {
+                    user.RaiseMessage(
+                        "\"{0}\" and \"{1}\" are the same versions.", leftVersion, rightVersion);
+                }
+                else if (compareResult == -1)
+                {
+                    user.RaiseMessage(
+                        "\"{0}\" is lower than \"{1}\".", leftVersion, rightVersion);
+                }
+                else if (compareResult == 1)
+                {
+                    user.RaiseMessage(
+                        "\"{0}\" is higher than \"{1}\".", leftVersion, rightVersion);
+                }
+                else
+                {
+                    user.RaiseMessage(
+                        "Usage: ckan compare version1 version2");
+                }
+            }
+            else
+            {
+                user.RaiseMessage(
+                    "Usage: ckan compare version1 version2");
+                return Exit.BADOPT;
+            }
 
-            return 0;
+            return Exit.OK;
         }
     }
 }

--- a/Cmdline/Main.cs
+++ b/Cmdline/Main.cs
@@ -217,7 +217,7 @@ This is a bad idea and there is absolutely no good reason to do it. Please run C
                     return repo.RunSubCommand((SubCommandOptions) cmdline.options);
 
                 case "compare":
-                    return (new Compare()).RunCommand(manager.CurrentInstance, cmdline.options);
+                    return (new Compare(user)).RunCommand(manager.CurrentInstance, cmdline.options);
 
                 default:
                     user.RaiseMessage("Unknown command, try --help");

